Effects/FlxFloodFill/source/FlxFloodFill.hx

```diff
@@ -0,0 +1,115 @@
+/**
+ * 
+ * A HaxeFlixel port of Photonstorm's 'FloodFillFX':
+ * https://github.com/photonstorm/Flixel-Power-Tools/blob/master/src/org/flixel/plugin/photonstorm/FX/FloodFillFX.as
+ * 
+ */
+
+package;
+
+import flash.display.BitmapData;
+import flash.geom.Point;
+import flash.geom.Rectangle;
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.util.FlxColor;
+import flixel.util.FlxDestroyUtil;
+
+/**
+ * "Creates a flood fill effect FlxSprite, useful for bringing in images in cool ways"
+ */
+class FlxFloodFill extends FlxSprite
+{
+	private var complete:Bool = false;
+	private var isFilling:Bool = false;
+	private var dropRect:Rectangle;
+	private var dropPoint:Point;
+	private var dropY:Int;
+	private var srcBitmapData:BitmapData;
+	private var fillDelay:Float = .05;
+	private var fillClock:Float = 0;
+	
+	/**
+	 * How many pixels to drop per update
+	 */
+	private var fillOffset:Int = 1;
+	
+	/**
+	 * @param x The effect's x-position.
+	 * @param x The effect's y-position.
+	 * @param srcBmd The image data used for the effect.
+	 * @param width The width of the effect.
+	 * @param height The height of the effect (a value larger than the source image makes the effect taller!).
+	 * @param fillLinesPerUpdate Number of lines per update to fill the effect with.
+	 * @param delayPerUpdate The time delay between each update.
+	 */
+	public function new(x:Float, y:Float, srcBmd:BitmapData, ?width:Int, ?height:Int, ?fillLinesPerUpdate:Int = 1, ?delayPerUpdate:Float = .05)
```

Comment:
I guess this was intentional for at least `width` and `height` - still not a good idea IMO, could just use a `== 0` check there.